src/leaflet_d3.js

```diff
@@ -23,32 +23,44 @@ L.CartoDBd3Layer = L.Class.extend({
 		var self = this;
 		options = options || {};
 		L.Util.setOptions(this, options);
+		this.providers = {
+			"windshaft": require('./providers/windshaft'),
+			"sql": require('./providers/sql')
+		}
 	},
 
 	onAdd: function (map) {
 		this._map = map;
 		this.options.map = map;
 		this.options.layer = this;
+
 		this.renderer = new Renderer(this.options);
+		this.provider = new this.providers.sql(this.options);
```

Comment:
I guess this is a temporal thing and it will be created within the constructor so a default provider can be choose but also one might be provided/configured.